### PR TITLE
add last login refresh request event listener

### PR DESCRIPTION
**Pull request by alexander-schranz**
| Q | A
| --- | ---
| Bug fix? | no
| New feature? | yes
| BC breaks? | no
| Deprecations? | no
| License | MIT

#### What's in this PR?

Add posibility to refresh last login every 10 minutes.

### CODE REVIEW

File: DependencyInjection/Configuration.php [hunk 1]
```diff
@@ -79,6 +83,12 @@ public function getConfigTreeBuilder()
 
         $rootNode
             ->children()
+                ->arrayNode(self::LAST_LOGIN)
```
Comment (wachterjohannes) [inline DependencyInjection/Configuration.php]: i would use `canBeEnabled` and set a default value to `refresh_interval`

File: EventListener/LastLoginListener.php [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+ */
+
+namespace Sulu\Bundle\CommunityBundle\EventListener;
+
+use Doctrine\ORM\EntityManager;
+use Sulu\Bundle\SecurityBundle\Entity\BaseUser;
+use Symfony\Component\HttpKernel\Event\FilterControllerEvent;
+use Symfony\Component\Security\Core\Authentication\Token\Storage\TokenStorageInterface;
+
+class LastLoginListener
```
Comment (wachterjohannes) [inline EventListener/LastLoginListener.php]: phpdoc

File: EventListener/LastLoginListener.php [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+        $this->entityManager = $entityManager;
+        $this->interval = (int) $interval;
+    }
+
+    /**
+     * Update the last login in specific interval.
+     *
+     * @param FilterControllerEvent $event
+     */
+    public function onCoreController(FilterControllerEvent $event)
```
Comment (wachterjohannes) [inline EventListener/LastLoginListener.php]: why not on `kernel.terminate`?

File: EventListener/LastLoginListener.php [hunk 4]
```diff
@@ -0,0 +1,92 @@
...
+    /**
+     * Check if user was active shortly.
+     *
+     * @param BaseUser $user
+     *
+     * @return bool
+     */
+    public function isActiveNow(BaseUser $user)
+    {
+        $delay = new \DateTime($this->interval . ' minutes ago');
```
Comment (wachterjohannes) [inline EventListener/LastLoginListener.php]: could you configure this string? `10 minutes` or `10 seconds` you could then configure it more accuate for your application. `10 minutes` seems a good default - but some faster applications maybe want to configure that in seconds or in hours for very slow apps.

File: UPGRADE.md [hunk 5]
```diff
@@ -1,5 +1,12 @@
 # Upgrade
 
+## 0.3.0
+
+### Parameter `sulu_community.config` was removed
+
+The whole config as parameter is not longer available the webspaces config
+you can get over the `sulu_community.webspaces` parameter.
```
Comment (wachterjohannes) [inline UPGRADE.md]: calling it `sulu_community.webspace_config` maybe?

Comment (alexander-schranz) [inline UPGRADE.md]: @wachterjohannes it is not a single webspace it contains all webspaces

File: EventListener/LastLoginListener.php (hunk 4 above)
Comment (alexander-schranz) [inline EventListener/LastLoginListener.php]: @wachterjohannes only want an integer value in the configuration but can set it to seconds so it like the cacheLifetime. ok?

File: EventListener/LastLoginListener.php (hunk 3 above)
Comment (alexander-schranz) [inline EventListener/LastLoginListener.php]: @wachterjohannes if I remember it correctly we did that if a query based on active users dont return 0 users when 1 user is online and the last login will be refreshed.

File: UPGRADE.md (hunk 5 above)
Comment (wachterjohannes) [inline UPGRADE.md]: `sulu_community.webspaces_config`?

File: EventListener/LastLoginListener.php (hunk 4 above)
Comment (wachterjohannes) [inline EventListener/LastLoginListener.php]: ok (:

File: DependencyInjection/Configuration.php (hunk 1 above)
Comment (alexander-schranz) [inline DependencyInjection/Configuration.php]: done

File: UPGRADE.md (hunk 5 above)
Comment (alexander-schranz) [inline UPGRADE.md]: done

**Comment by alexander-schranz** [comment]
@wachterjohannes ready

**Comment by wachterjohannes** [comment]
@alexander-schranz will wait for tests and then merging it.

**Pull request merged**